Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1159,7 +1159,7 @@ public extension Purchases {
 #endif
 
     @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
-    func handleObserverModeTransaction(
+    func didCompletePurchaseWithResult(
```

Comment:
I believe that we wanted to change this function's name to `handlePurchaseResult` (PR [here](https://github.com/RevenueCat/purchases-ios/pull/3950)), can we keep this named `handleObserverModeTransaction` here and use the other PR to rename this function?

We could also merge that PR into this branch instead of `5.0-dev` if you'd like to keep all the interface changes in one branch!

Reply:
As discussed in Slack, we may change this method to some variant of `registerPurchaseResult`, tbd.